src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -36,12 +38,34 @@
 
         :else (app req)))))
 
+(defn- blocking-submit-command [do-submit-fn response-pub uuid completion-timeout-ms]
+  (let [response-chan (async/chan)
+        topic (str uuid)
+        _ (async/sub response-pub topic response-chan)
+        timeout-chan (async/timeout completion-timeout-ms)
+        _ (do-submit-fn)
+        timed-out (async/alt!! timeout-chan true
+                               response-chan false)]
+    (async/unsub response-pub topic response-chan)
+    (http/json-response {:uuid uuid, :processed (not timed-out)}
+                        (if timed-out 503 200))))
+
```

Comment:
I think I grok what's going on here but don't know as much about this in core.async. Is there a need to put a close call for the channel and maybe the unsub in a finally clause? The sub call will close the response-chan when the message is published, but what if it doesn't get published? Does response-chan get GC'd? Or if there's a failure and the unsub doesn't get called?